controllers/component/component_adapter.go

```diff
@@ -55,6 +58,11 @@ func NewAdapter(component *applicationapiv1alpha1.Component, application *applic
 // EnsureComponentIsCleanedUp is an operation that will ensure components
 // marked for deletion have a snapshot created without said component
 func (a *Adapter) EnsureComponentIsCleanedUp() (controller.OperationResult, error) {
+	if !controllerutil.ContainsFinalizer(a.component, h.ComponentFinalizer) {
```

Comment:
I'm thinking that this might be too late to add a finalizer to a resource, since it's already being deleted.

An alternative solution might be to not try and `Get` the component from the cluster in the component_controller.go and instead use the `req.NamespacedName.Name` to extract the `Name` of the component and use that for the further logic in this adapter.

Reply:
Just realized that my alternative suggestion would be missing a way to find the `Application`, but my original point still stands.

Maybe we can also react to `Component` creation events and make sure that the finalizer is added then?